.github/workflows/check-reference.yml

```
@@ -0,0 +1,39 @@
+name: Check patch references
+
+on:
+  push:
+    branches-ignore:
+      - dependabot/**
+  schedule:
+    # Once every day at midnight UTC
+    - cron: "0 0 * * *"
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout patches
+        uses: actions/checkout@v4
+
+      - name: Checkout core
+        uses: actions/checkout@v4
+        with:
+          repository: 'homebrew/homebrew-core'
+          path: homebrew-core
+
+      - name: Detect references
+        run: |
+          #!/bin/bash
+
+          core_patches=$(git -C homebrew-core grep -h "Homebrew/formula-patches" | awk -F/ '{print $NF}' | tr -d '"')
```

```suggestion
          core_patches=$(git -C homebrew-core grep -h "Homebrew/formula-patches" | awk -F/ '{print $(NF-1)"/"$NF}' | tr -d '"')
```

You want the subdirectory too (though this strongly assumes no further nesting).